src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -467,23 +461,71 @@ private void LookupExtensionMethodsInSingleBinder(ExtensionMethodScope scope, Lo
             else if (resultWithoutSuffixIsViable)
             {
                 // single viable lookup symbol only found without Attribute suffix, return result.
+                if (!Compilation.LanguageVersion.AllowGenericAttributes())
+                {
+                    var noAliasSymbolWithoutSuffix = (NamedTypeSymbol)UnwrapAliasNoDiagnostics(symbolWithoutSuffix);
+                    if (noAliasSymbolWithoutSuffix.IsGenericType)
+                    {
+                        var diagInfo = new CSDiagnosticInfo(ErrorCode.ERR_AttributeCantBeGeneric, noAliasSymbolWithoutSuffix);
+                        result.SetFrom(GenerateNonViableAttributeTypeResult(noAliasSymbolWithoutSuffix, diagInfo));
+                    }
+                }
             }
             else if (resultWithSuffixIsViable)
             {
                 Debug.Assert(resultWithSuffix != null);
 
-                // Single viable lookup symbol only found with Attribute suffix, return resultWithSuffix.
-                result.SetFrom(resultWithSuffix);
+                if (!Compilation.LanguageVersion.AllowGenericAttributes())
+                {
+                    var noAliasSymbolWithSuffix = (NamedTypeSymbol)UnwrapAliasNoDiagnostics(symbolWithSuffix);
+                    if (noAliasSymbolWithSuffix.IsGenericType)
+                    {
+                        var diagInfo = new CSDiagnosticInfo(ErrorCode.ERR_AttributeCantBeGeneric, noAliasSymbolWithSuffix);
+                        result.SetFrom(GenerateNonViableAttributeTypeResult(noAliasSymbolWithSuffix, diagInfo));
+                    }
+                    else
+                    {
+                        // Single viable lookup symbol only found with Attribute suffix, return resultWithSuffix.
+                        result.SetFrom(resultWithSuffix);
+                    }
+                }
+                else
+                {
+                    // Single viable lookup symbol only found with Attribute suffix, return resultWithSuffix.
+                    result.SetFrom(resultWithSuffix);
+                }
             }
             else
             {
+                if (!Compilation.LanguageVersion.AllowGenericAttributes())
+                {
+                    if (symbolWithoutSuffix != null)
+                    {
+                        var noAliasSymbolWithoutSuffix = UnwrapAliasNoDiagnostics(symbolWithoutSuffix);
+                        if (noAliasSymbolWithoutSuffix.Kind == SymbolKind.NamedType && ((NamedTypeSymbol)noAliasSymbolWithoutSuffix).IsGenericType)
```

Comment:
nit: could use pattern `noAliasSymbolWithoutSuffix is NamedTypeSymbol { IsGenericType: true }`